libraries/Microsoft.Bot.Builder.Dialogs.Declarative/DialogComponentRegistration.cs

```diff
@@ -15,6 +15,7 @@ public override IEnumerable<JsonConverter> GetConverters(ISourceMap sourceMap, I
             yield return new InterfaceConverter<Dialog>(refResolver, sourceMap, paths);
             yield return new InterfaceConverter<IStorage>(refResolver, sourceMap, paths);
             yield return new InterfaceConverter<IRecognizer>(refResolver, sourceMap, paths);
```

Comment:
>IRecognizer [](start = 48, length = 11)

Do we want to deprecate IRecognizer?

Reply:
I considered it, but there is nothing inherently wrong with it and people might be using it outside of our dialog system..  It's just AdaptiveDialog will use the new Recognizer class.

---
In reply to: [360621755](https://github.com/microsoft/botbuilder-dotnet/pull/3174#discussion_r360621755) [](ancestors = 360621755)